crates/noirc_evaluator/src/ssa/optim.rs

```diff
@@ -123,6 +124,19 @@ pub fn simplify(ctx: &mut SsaContext, ins: &mut node::Instruction) {
                     return;
                 }
             }
+            node::Operation::Shl => {
+                ins.operator = node::Operation::Mul;
+                //todo checks that 2^rhs does not overflow
+                ins.rhs = ctx.get_or_create_const(FieldElement::from(2_i128).pow(&r_const), r_type);
+                return;
+            }
+            node::Operation::Shr => {
+                assert!(matches!(ins.res_type, node::ObjectType::Unsigned(_)));
+                ins.operator = node::Operation::Udiv;
```

Comment:
Why are we asserting that the result of a `>>` must always be an unsigned type?

Reply:
Because the behavior for signed integer is different and not yet implemented, may be I should change this into a todo!()